src/super_gradients/training/utils/callbacks.py

```diff
@@ -195,30 +197,76 @@ def __init__(self, email, model_meta_data, optimization_request_form, password=N
         password = password or getpass.getpass()
         self.platform_client.login(email, password)
 
+    @staticmethod
+    def log_optimization_failed():
+        logger.info(f"We couldn't finish your model optimization. Visit https://console.deci.ai for details")
+
+    def upload_model(self, model):
+        """
+            This function will upload the trained model to the Deci Lab
+
+            Args:
+                model: The resulting model from the training process
+        """
+        self.platform_client.add_model(
+            add_model_request=self.model_meta_data,
+            optimization_request=self.optimization_request_form,
+            local_loaded_model=model.module.cpu(),
+            **self.conversion_kwargs
+        )
+
+    def get_optimization_status(self, optimized_model_name: str):
+        """
+            This function will do fetch the optimized version of the trained model and check on its benchmark status.
+            The status will be checked against the server every 30 seconds and the process will timeout after 30 minutes
+            or log about the successful optimization - whichever happens first.
+            Args:
+                optimized_model_name (str): Optimized model name
+            Returns:
+                bool: whether or not the optimized model has been benchmarked
+        """
+        def handler(_signum, _frame):
+            logger.error('Process timed out. Visit https://console.deci.ai for details')
+            return False
+
+        signal.signal(signal.SIGALRM, handler)
+        signal.alarm(1800)
+
+        finished = False
+        while not finished:
+            optimized_model = self.platform_client.get_model_by_name(name=optimized_model_name).data
+            if optimized_model.benchmark_state not in [ModelBenchmarkState.IN_PROGRESS, ModelBenchmarkState.PENDING]:
+                finished = True
+            else:
+                time.sleep(30)
+
+        signal.alarm(0)
+        return True
+
     def __call__(self, context: PhaseContext):
+        """
+            This function will attempt to upload the trained model and schedule an optimization for it.
+            Args:
+                context (PhaseContext): Training phase context
+            Returns:
+                bool: whether or not the optimized model has been benchmarked
+            """
         try:
             model = copy.deepcopy(context.net)
-            model_state_dict_path = os.path.join(context.ckpt_dir, self.ckpt_name)['net']
-            model.load_state_dict(model_state_dict_path)
```

Comment:
This line shouldn't be removed.  There's a a reason we load the weights from model_state_dict_path-
The callback is fired at the end of training and the weights at that point might not be optimal (or just not the ones we meant when specifying them in ckpt_name.